provider/cluster_waiter_resource.go

```diff
@@ -179,3 +213,23 @@ func (r *ClusterWaiterResource) retryClusterReadiness(attempts int, sleep time.D
 
 	return object, nil
 }
+
+func timeoutValidators() []tfsdk.AttributeValidator {
+	errSumm := "Invalid timeout configuration"
+	return []tfsdk.AttributeValidator{
+		&common.AttributeValidator{
+			Desc: "Timeout must be a positive number",
+			Validator: func(ctx context.Context, req tfsdk.ValidateAttributeRequest, resp *tfsdk.ValidateAttributeResponse) {
+				timeout := &types.Int64{}
+				diag := req.Config.GetAttribute(ctx, req.AttributePath, timeout)
+				if diag.HasError() {
+					// No attribute to validate
+					return
+				}
+				if !timeout.Null && !timeout.Unknown && timeout.Value < 0 {
```

Comment:
Can it be zero?
The error message below indicates that the answer is no :-)
```suggestion
				if !timeout.Null && !timeout.Unknown && timeout.Value <= 0 {
```